include/bluetooth/gatt.h

```diff
@@ -746,35 +769,51 @@ ssize_t bt_gatt_attr_read_cpf(struct bt_conn *conn,
  */
 typedef void (*bt_gatt_complete_func_t) (struct bt_conn *conn, void *user_data);
 
-/** @brief Notify attribute value change with callback.
+struct bt_gatt_notify_params {
+	/** Notification Attribute UUID type */
+	const struct bt_uuid *uuid;
+	/** Notification Attribute object*/
+	const struct bt_gatt_attr *attr;
+	/** Notification Value data */
+	const void *data;
+	/** Notification Value length */
+	u16_t len;
+	/** Notification Value callback */
```

Comment:
This should clearly document the context (thread) that the callback will be called from. We should probably do a review of all our APIs to make sure it's clearly documented what context various callbacks get called from.

Reply:
Right will add the context of the callbacks.

Reply:
Fixed

Reply:
Wouldn't it be better to have a mention of it here as well, since that's likely the first place a developer will look. In fact, this probably deserves a bit more explanation about when exactly the callback happens (i.e. what it means).